src/server.c

```diff
@@ -1394,6 +1394,9 @@ static int on_client_enable_continuous_updates(struct nvnc_client* client)
 	client->continuous_updates_enabled = msg->enable_flag;
 
 	if (msg->enable_flag) {
+		/* If there are any pending messages left, make sure the are processed */
+		process_fb_update_requests(client);
```

Comment:
This really should come after `client->continuous_updates` has been populated.

Reply:
I moved it.